realm/realm-library/src/main/java/io/realm/DynamicRealmObject.java

```diff
@@ -650,6 +658,18 @@ public RealmFieldType getFieldType(String fieldName) {
         return proxyState.getRow$realm().getColumnType(columnIndex);
     }
 
+    private void checkFieldType(String fieldName, long columnIndex, RealmFieldType expectedType) {
+        RealmFieldType columnType = proxyState.getRow$realm().getColumnType(columnIndex);
+        if (columnType != expectedType) {
+            String indefiniteVowel = "";
+            if (expectedType == RealmFieldType.INTEGER || expectedType == RealmFieldType.LIST) {
```

Comment:
`expectedType == RealmFieldType.LIST` should be `expectedType == RealmFieldType.OBJECT`?